tests/test_viewer.py

```diff
@@ -373,20 +373,14 @@ def test_directory_flamegraph(self):
     def test_directory_timeout(self):
         test_data_dir = os.path.join(os.path.dirname(__file__), "data")
         with Viewer(test_data_dir, timeout=2):
-            time.sleep(0.5)
-            resp = urllib.request.urlopen("http://127.0.0.1:9001/")
-            self.assertEqual(resp.code, 200)
-            self.assertIn("fib.json", resp.read().decode("utf-8"))
-            resp = urllib.request.urlopen("http://127.0.0.1:9001/fib.json")
-            self.assertEqual(resp.url, "http://127.0.0.1:9002/")
+            self.basic_judgment()
             time.sleep(2.5)
             resp = urllib.request.urlopen("http://127.0.0.1:9001/old.json")
             self.assertEqual(resp.url, "http://127.0.0.1:9002/")
 
     @unittest.skipIf(sys.platform == "win32", "Can't send Ctrl+C reliably on Windows")
     def test_directory_max_port(self):
-        try:
-            tmp_dir = tempfile.mkdtemp()
+        with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
I do remember that I explicitly use try .. finally.. here instead of `TemporaryDirectory`. After some second thoughts, this might not be necessary. So this change might be correct.

Reply:
Yes, I was a bit confused when I saw it. If there is a bug about `tempfile.mkdtemp()`, then there will be no `finish`, but it won't fail I guess, then there is no need to use try...finally...?, so I wanted to put `tmp_dir = tempfile.mkdtemp()` in front of `try` at first.